portal-api/src/v1/users/controller.js

```diff
@@ -93,23 +94,30 @@ exports.list = async (callback) => {
   collection.find().toArray(callback);
 };
 
+/**
+ * @deprecated Use findById inside user repository instead
+ */
 exports.findOne = async (_id, callback) => {
-  if (!ObjectId.isValid(_id)) {
-    throw new Error('Invalid User Id');
+  const userRepository = new UserRepository();
+  try {
+    const user = await userRepository.findById(_id);
+    callback(null, user);
+  } catch (error) {
+    callback(error, null);
   }
-
-  const collection = await db.getCollection('users');
-
-  collection.findOne({ _id: { $eq: ObjectId(_id) } }, callback);
 };
 
+/**
+ * @deprecated Use findByUsername inside user repository instead
+ */
 exports.findByUsername = async (username, callback) => {
```

Comment:
The error-case behaviours for `findOne` and `findByUsername` have changed from this - have we been able to check that the differences are safe? :)

Reply:
reverted this back